client/src/Utils/Contacts/ContactIdValidationSchema.ts

```diff
@@ -1,26 +1,34 @@
 import * as yup from 'yup';
 
-import { ID_BASIC_VALIDATION_REGEX } from 'commons/Regex/Regex';
-import IdentificationTypes from 'models/enums/IdentificationTypes';
+import IdentificationTypesCodes from 'models/enums/IdentificationTypesCodes';
 import InteractionEventContactFields from 'models/enums/InteractionsEventDialogContext/InteractionEventContactFields';
-import { isIdValid , isPassportValid, idLength } from 'Utils/auxiliaryFunctions/auxiliaryFunctions';
-import { invalidIdText, invalidPassportText } from 'commons/Schema/messages';
+import { invalidIdText, invalidOtherIdText, invalidPalestineIdText, invalidPassportText } from 'commons/Schema/messages';
+import { isIdValid , isPassportValid, isPalestineIdValid, isOtherIdValid } from 'Utils/auxiliaryFunctions/auxiliaryFunctions';
 
-const ContactIdValidationSchema = yup
-  .string()
-  .when(InteractionEventContactFields.IDENTIFICATION_TYPE, {
-      is: IdentificationTypes.PASSPORT,
-      then: yup
-        .string()
-        .nullable()
-        .test('isValid', invalidPassportText, (id) => isPassportValid(id)),
-      otherwise:
-        yup
-        .string()
-        .nullable()
-        .matches(ID_BASIC_VALIDATION_REGEX, 'ת.ז חייבת להכיל ספרות בלבד')
-        .length(idLength, `ת.ז מכילה ${idLength} ספרות בלבד`)
-        .test('isValid', invalidIdText, (id) => isIdValid(id)),
+const ContactIdValidationSchema = yup.string()
+    .when(InteractionEventContactFields.IDENTIFICATION_TYPE, {
+        is: IdentificationTypesCodes.ID,
+        then: yup.string()
+            .nullable()
+            .test('isValid', invalidIdText, (id) => isIdValid(id)),
+        otherwise:
+            yup.string().when(InteractionEventContactFields.IDENTIFICATION_TYPE, {
+            is: IdentificationTypesCodes.PASSPORT,
+            then: yup.string()
+                .nullable()
+                .test('isValid', invalidPassportText, (id) => isPassportValid(id)),
+            otherwise:
+                yup.string().when(InteractionEventContactFields.IDENTIFICATION_TYPE, {
+                is: IdentificationTypesCodes.PALESTINE_ID,
+                then: yup.string()
+                    .nullable()
+                    .test('isValid', invalidPalestineIdText, (id) => isPalestineIdValid(id)),
+                otherwise:
+                    yup.string()
+                    .nullable()
+                    .test('isValid', invalidOtherIdText, (id) => isOtherIdValid(id)),
+            })
+        })
```

Comment:
maybe It's more effective to use when as a function - see examples in [here](https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema)

Reply:
I'm using when the problem is that I have to compare to various values so more than one when is needed.

Reply:
yea but you can use a function as the second handler - look at the link that I refrenced

Reply:
changed it according to the example